### DISCUSSION TITLE
How to delete non-relevant commits

### DISCUSSION THREAD

**Discussion by sanderfoobar** [community/community#22888] [New to GitHub]
<p>Question regarding commits that float around after an --amend:</p>
<ol>
<li>
<p>Push a commit with a reference to an issue:</p>
<p>git init<br>
git add foo.py<br>
git add README.md<br>
git commit -m “Fixes a thing. Issue reference: <span class="hashtag">#164</span>”<br>
git push origin my-cool-pr</p>
</li>
<li>
<p>Realizing you forgot something, you --amend:</p>
<p>git add bar.py<br>
git commit -a --amend<br>
git push origin my-cool-pr -f</p>
</li>
</ol>
<p>3. It’s not your day, you made a spelling mistake in README.md</p>
<pre><code>vim README.md
git add README.md
git commit -a --amend
git push origin my-cool-pr -f
</code></pre>
<p>Now visit <em>issue <span class="hashtag">#164</span></em>. It will have 3 ‘back-references’ to those individual commits. However, only the last one is relevant, since you’ve been amending. </p>
<p>How do I delete those commits that are still floating around?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>There isn’t a way for you to manually delete those commits. You would have to contact private support at <a href="https://github.com/contact" rel="nofollow noopener">https://github.com/contact</a> to have them removed.</p>
<p>For this reason, I typically don’t recommend adding issue references in commit messages themselves. Rather, I recommend adding the issue references to the PR body when a series of commits addresses or fixes an issue.</p>
<p>I hope that helps!</p>
